x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/analytics/action/support/BaseAnalyticsCollectionResponse.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.analytics.action.support;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.entsearch.analytics.AnalyticsCollection;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class BaseAnalyticsCollectionResponse extends ActionResponse {
```

Comment:
Good call capturing common code in this class! If PUT action does not need to return analytics information, then it might be unnecessary.

Reply:
Yeh its an interesting point. Right now i've kept it to share the same response model as previous API did so. I think the base response class will come useful more when LIST is implemented too. Also potentially in future an analytics collection may be made up from a couple of service calls (when we get into sessions and Users).